### PR TITLE
Display informational version

### LINKED ISSUE #918: Dev versions of MSBuild should put the version with commit ID in the logo
We get banners like

```
Microsoft (R) Build Engine version 15.1.188.30335
Copyright (C) Microsoft Corporation. All rights reserved.
```

But it'd be really nice (for like 8 people) if that had the full form of the version that includes the git commit ID for easy checkout.

### PR DESCRIPTION
**Pull request by jeffkl**
Closes #918

### CODE REVIEW

**Comment by jeffkl** [comment]
Sample:

```
Microsoft (R) Build Engine version 15.8.60-preview+gb69bce195d for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

15.8.60-preview+gb69bce195d
```

File: src/MSBuild/XMake.cs [hunk 1]
```diff
@@ -3476,7 +3476,7 @@ private static void DisplayCopyrightMessage()
             const string frameworkName = ".NET Framework";
 #endif
 
-            Console.WriteLine(ResourceUtilities.FormatResourceString("CopyrightMessage", ProjectCollection.Version.ToString(), frameworkName));
+            Console.WriteLine(ResourceUtilities.FormatResourceString("CopyrightMessage", ThisAssembly.AssemblyInformationalVersion, frameworkName));
```
Comment (rainersigwald) [inline src/MSBuild/XMake.cs]: 🎉

File: src/MSBuild/XMake.cs [hunk 2]
```diff
@@ -3539,7 +3539,7 @@ private static void ShowHelpPrompt()
         /// </summary>
         private static void ShowVersion()
         {
-            Console.Write(ProjectCollection.Version.ToString());
+            Console.WriteLine(ThisAssembly.AssemblyInformationalVersion);
```
Comment (rainersigwald) [inline src/MSBuild/XMake.cs]: This one I'm more worried about. Think people are parsing the output of `msbuild /nologo /version` and will be broken by a non-`System.Version` compatible output? I suspect they might :(

Comment (jeffkl) [reply src/MSBuild/XMake.cs]: agreed

File: src/MSBuild/XMake.cs (hunk 1 above)
Comment (cdmihai) [reply src/MSBuild/XMake.cs]: `ThisAssembly` is not available during sourdebuild builds (as git versioning is turned off). There's a `THISASSEMBLY` symbol you can ifdef on :(

Comment (jeffkl) [reply src/MSBuild/XMake.cs]: Added

**Pull request merged**